scripts/count_opcodes_zkSync.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+ 
+# Script to create report on bytecodes not supported by zkSync
+# https://github.com/DistributedCollective/Sovryn-smart-contracts/issues/285
+
+import json
+import re
+import csv
+
+# Hardhat json packs opcodes in 1 line. Better visual on multiline.
+def opcode1linerToMultiline(oneLiner):
+    oneLiner = re.sub(r' ', r' ___NEWLINE ', oneLiner)
+    oneLiner = re.sub(r' ___NEWLINE (0x[^ ]+)', r' \1', oneLiner)
+    oneLiner = re.sub(r' ___NEWLINE ', r'\n', oneLiner)
+    return oneLiner
+
+# Get the frequency of a given code
+def countCode(code, opcode):
+    return opcode.count(code)
+
+# The sourcemap format is compressed, so a decoder is needed.
+# Specification of the format is in the solidity documentation:
+# https://docs.soliditylang.org/en/v0.7.5/internals/source_mappings.html
+# This function maps instruction indexes to source offsets.
+# Deprecated: even though decoding were performed, there is not a
+# one-to-one matching of sourcemap references to opcodes. For example,
+# a simple contract has 21 references in the sourceMap (20 semicolons)
+# for an opcode set of near 100 instructions. So it is useless to follow
+# this route in order to trace opcodes to the source with code line accuracy.
+# def decodeSourceMap(srcmap):
+#     print ("decodeSourceMap:")
+#     print ("\nsrcmap: " + srcmap)
+#     s = srcmap.split(";")
+#     print ("\nsplitted: " + ' '.join(s))
+#     exit()
+
+# Opening the JSON output from the last hardhat compilation
+# f = open('artifacts/build-info/eaaf433b5af1a78be582155342f51ab5.json',)
+f = open('artifacts/build-info/0202910f40424d4ba1f4d5c762daecf5.json',)
```

Comment:
this should not be hardcoded, please use params
and no artifacts/ should be merged

Reply:
Fixed